bundle/src/test/java/com/adobe/acs/commons/wcm/impl/PublishUrlServletTest.java

```diff
@@ -0,0 +1,88 @@
+ * #L%
+ */
+
+package com.adobe.acs.commons.wcm.impl;
+
+import com.day.cq.commons.Externalizer;
+import io.wcm.testing.mock.aem.junit5.AemContext;
+import io.wcm.testing.mock.aem.junit5.AemContextExtension;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletRequest;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletResponse;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.servlet.Servlet;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith({MockitoExtension.class, AemContextExtension.class})
+class PublishUrlServletTest {
+
+    private final AemContext ctx = new AemContext();
+
+    @Mock
+    Externalizer externalizer;
+
+    @Test
+    void testDoGet() throws IOException {
+        ctx.registerAdapter(ResourceResolver.class, Externalizer.class, externalizer);
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("externalizerKeys", new String[]{"local", "author", "publish", "dispatcher"});
+
+        Map<String, Object> requestParams = new HashMap<>();
+        requestParams.put("path", "/content/we-retail/us/en/experience");
+        ctx.request().setParameterMap(requestParams);
+
+        //when(config.externalizerKeys()).thenReturn(new String[]{"local", "author", "publish", "dispatcher"});
+        when(externalizer.externalLink(any(ResourceResolver.class), any(String.class), any(String.class), any(String.class)))
+                .thenReturn("http://localhost:4502/content/we-retail/us/en/experience.html")
+                .thenReturn("https://aem.author.someorganization.com/content/we-retail/us/en/experience.html")
+                .thenReturn("https://aem.publish.someorganization.com/content/we-retail/us/en/experience.html")
+                .thenReturn("https://www.someorganization.com/experience");
+
+        ctx.registerInjectActivateService(new PublishUrlServlet(), params);
+        
+        PublishUrlServlet servlet = (PublishUrlServlet) ctx.getServices(Servlet.class,
+                "(sling.servlet.resourceTypes=acs-commons/components/utilities/publish-url)")[0];
+
+        servlet.doGet(ctx.request(), ctx.response());
+
+        try (InputStream inputStream = getClass().getResourceAsStream("PublishUrlServletResponse.json")) {
+            assert inputStream != null;
+            String expectedJson = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+            assertEquals(expectedJson, ctx.response().getOutputAsString());
+        }
+    }
```

Comment:
Just a FYI - I updated this to use AEM Mocks and simulate more "real" code execution rather than mocking everything.

Reply:
Thanks, and that makes sense. Simplify when you can.